devices/Bmm150/Bmm150.cs

```diff
@@ -0,0 +1,359 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.Device.Model;
+using System.Diagnostics;
+using System.IO;
+using System.Numerics;
+using System.Threading;
+
+namespace Iot.Device.Magnetometer
+{
+    /// <summary>
+    /// Bmm150 class implementing a magnetometer
+    /// </summary>
+    [Interface("Bmm150 class implementing a magnetometer")]
+    public sealed class Bmm150 : IDisposable
+    {
+        /// <summary>
+        /// I2c device comm channel
+        /// </summary>
+        private I2cDevice _i2cDevice;
+
+        /// <summary>
+        /// Bmm150 device interface
+        /// </summary>
+        private Bmm150I2cBase _Bmm150Interface;
+
+        /// <summary>
+        /// Bmm150 Trim extended register data
+        /// </summary>
+        private Bmm150TrimRegisterData _trimData;
+
+        /// <summary>
+        /// Magnetometer calibration compensation vector
+        /// </summary>
+        private Vector3 _calibrationCompensation;
+
+        /// <summary>
+        /// Magnetometer (R-HALL) temperature compensation value, used in axis compensation calculation functions
+        /// </summary>
+        private uint _rHall;
+
+        /// <summary>
+        /// Flag to evaluate disposal of resources
+        /// </summary>
+        private bool _shouldDispose = true;
+        
+        /// <summary>
+        /// Default I2C address for the Bmm150
+        /// In the official sheet (P36) states that address is 0x13, alhtough for m5stack is 0x10
+        /// more info: https://github.com/m5stack/M5_BMM150/blob/master/src/M5_BMM150_DEFS.h#L163
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x10;
+
+        /// <summary>
+        /// Default timeout to use when timeout is not provided in the reading methods
+        /// </summary>
+        [Property]
+        public TimeSpan DefaultTimeout { get; set; } = TimeSpan.FromSeconds(1);
+
+        /// <summary>
+        /// Default constructor for an independent Bmm150
+        /// </summary>
+        /// <param name="i2CDevice">The I2C device</param>
+        public Bmm150(I2cDevice i2CDevice)
+            : this(i2CDevice, new Bmm150I2c())
+        {
+        }
+
+        /// <summary>
+        /// Constructor to use if Bmm150 is behind another element and need a special I2C protocol like
+        /// when used with the MPU9250
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device</param>
+        /// <param name="Bmm150Interface">The specific interface to communicate with the Bmm150</param>
+        /// <param name="shouldDispose">True to dispose the I2C device when class is disposed</param>
+        public Bmm150(I2cDevice i2cDevice, Bmm150I2cBase Bmm150Interface, bool shouldDispose = true)
```

Comment:
Not sure about the value of having this specific constructor.

Reply:
As discussed, we keep this interface. Therefore it makes sense exposing another constructor to enable devs to pass whatever interface they wish.